vmm/src/acpi.rs

```diff
@@ -626,81 +628,84 @@ fn create_viot_table(iommu_bdf: &PciBdf, devices_bdf: &[PciBdf]) -> Sdt {
     viot
 }
 
-pub fn create_acpi_tables(
-    guest_mem: &GuestMemoryMmap,
+fn create_acpi_tables_internal(
+    dsdt_offset: GuestAddress,
     device_manager: &Arc<Mutex<DeviceManager>>,
     cpu_manager: &Arc<Mutex<CpuManager>>,
     memory_manager: &Arc<Mutex<MemoryManager>>,
     numa_nodes: &NumaNodes,
     tpm_enabled: bool,
-) -> GuestAddress {
+    #[cfg(feature = "fw_cfg")] add_to_fw_cfg: bool,
+) -> (Rsdp, Vec<u8>) {
     trace_scoped!("create_acpi_tables");
-
-    let start_time = Instant::now();
-    let rsdp_offset = arch::layout::RSDP_POINTER;
     let mut tables: Vec<u64> = Vec::new();
 
+    // To create AcpiTable
+    let mut table_bytes: Vec<u8> = vec![];
+    #[cfg(feature = "fw_cfg")]
+    let mut pointers: Vec<usize> = vec![];
```

Comment:
IIUC, with `pointers`, what you need is pretty much the offset/address of all tables that XSDT points to, e.g. what `tables` does. Why can't we just reuse `tables`?

I see we need extra address/offset, say `pointer_facp_to_dsdt`, which is fine to be added to `tables` when needed.